src/v/cluster/persisted_stm.cc

```diff
@@ -164,11 +164,62 @@ ss::future<> persisted_stm::wait_offset_committed(
     return _c->commit_index_updated().wait(timeout, stop_cond);
 }
 
+ss::future<bool> persisted_stm::sync(
+  model::timeout_clock::duration timeout,
+  model::offset offset,
+  model::term_id term) {
+    auto committed = _c->committed_offset();
+    auto is_synced = false;
+    auto ntp = _c->ntp();
+
+    if (offset > committed) {
+        try {
+            co_await wait_offset_committed(timeout, offset, term);
+        } catch (...) {
+            vlog(
+              clusterlog.error,
+              "sync error: wait_offset_committed failed with {}; offsets: "
+              "dirty={}, committed={}; ntp={}",
+              std::current_exception(),
+              offset,
+              committed,
+              ntp);
+            co_return false;
+        }
+    } else {
+        offset = committed;
+    }
+
+    if (_c->term() == term) {
+        try {
+            co_await wait(offset, model::timeout_clock::now() + timeout);
+        } catch (...) {
+            vlog(
+              clusterlog.error,
+              "sync error: waiting for offset={} failed with {}; committed "
+              "offset={}; ntp={}",
+              offset,
+              std::current_exception(),
+              committed,
+              ntp);
+            co_return false;
+        }
+        is_synced = true;
+    }
+
+    if (is_synced) {
```

Comment:
it seems that this if statement is not required, it can be done under `if (_c->term() == term)` or maybe it would even be better to early return if term doesn't match like:
```
if(_c->term() != term){
  return false;
}

Reply:
good idea! I've changed it